### ISSUE TITLE
PR #3303 breaks build for non x86_64 architectures

### ISSUE THREAD

**Issue by m2049r** [monero-project/monero#3393]
```crypto/chacha.h``` calls ```cn_slow_hash_pre()``` from ```crypto/slow_hash.c``` but this method is only compiled for
```C
#if !defined NO_AES && (defined(__x86_64__) || (defined(_MSC_VER) && defined(_WIN64)))
```
and so, builds for arm & x86 fail with ```error: undefined reference to 'cn_slow_hash_pre'```

**Comment by stoffu** [comment]
Fixed by #3350

**Comment by moneromooo-monero** [comment]
+resolved
